### PR TITLE
Add channels.SetDefault and tests

### PR DESCRIPTION
**Pull request by sistason**
Add set_default normally.

I broke the reproducibility of the tests with my Oauth-test before, so this also fixes this.

### CODE REVIEW

**Comment by sistason** [comment]
(I'm used to gitlab, can you squash this via Github or do I need to do this locally?)

File: rocketchat_API/APISections/channels.py [hunk 1]
```diff
@@ -137,6 +137,15 @@ def channels_rename(self, room_id, name, **kwargs):
             "channels.rename", roomId=room_id, name=name, kwargs=kwargs
         )
 
+    def channels_set_default(self, room_id, default, **kwargs):
+        """Sets the description for the channel."""
```
Comment (jadolg) [inline rocketchat_API/APISections/channels.py:141]:
```suggestion
        """Sets whether the channel is a default channel or not."""
```

File: tests/test_channels.py [hunk 2]
```diff
@@ -380,3 +380,12 @@ def test_channels_online(logged_rocket):
     channels_online = logged_rocket.channels_online(query={"_id": "GENERAL"}).json()
     assert channels_online.get("success")
     assert len(channels_online.get("online")) >= 1
+
+
+def test_channels_set_default(logged_rocket):
+    results = logged_rocket.channels_set_default(
```
Comment (jadolg) [inline tests/test_channels.py:386]:
```suggestion
    channels_set_default = logged_rocket.channels_set_default(
```

File: tests/test_channels.py [hunk 3]
```diff
@@ -380,3 +380,12 @@ def test_channels_online(logged_rocket):
...
+        room_id="GENERAL", default=False
+    ).json()
+    assert results.get("channel").get("default") is False
+    results = logged_rocket.channels_set_default(room_id="GENERAL", default=True).json()
```
Comment (jadolg) [inline tests/test_channels.py:390]:
```suggestion
    channels_set_default = logged_rocket.channels_set_default(room_id="GENERAL", default=True).json()
```

Comment (jadolg) [reply tests/test_channels.py:390]: nit: In general, I always test if the method succeeded to make sure it fails fast.

File: tests/test_settings.py [hunk 4]
```diff
@@ -20,9 +20,22 @@ def test_settings_public(rocket):
...
+    logged_rocket.call_api_post(
+        "method.call/refreshOAuthService",
+        message='{"method": "refreshOAuthService", "params": []}',
+    )
     oauth_get = logged_rocket.settings_oauth().json()
     assert oauth_get.get("success")
-    assert not oauth_get.get("services")
+    if oauth_get.get("services"):
+        # remove the OAuth app Test beforehand, when this is not the first test run (for reproducibility)
+        logged_rocket.call_api_post(
```
Comment (jadolg) [inline tests/test_settings.py:32]: Is it possible to test this?

**Review by jadolg — CHANGES_REQUESTED** [review]

File: rocketchat_API/APISections/channels.py (hunk 1 above)
Comment (sistason) [reply rocketchat_API/APISections/channels.py]: fixed in 1a3458c

File: tests/test_channels.py (hunk 2 above)
Comment (sistason) [reply tests/test_channels.py]: fixed in 1a3458c

File: tests/test_channels.py (hunk 3 above)
Comment (sistason) [reply tests/test_channels.py]: I think this is now done, right?

File: tests/test_settings.py (hunk 4 above)
Comment (sistason) [reply tests/test_settings.py]: as it is a normal requests.post, I check the requests.Response in c99c308. You think this suffices?

**Comment by sistason** [comment]
sorry I totally forgot about this PR... :/

File: tests/test_settings.py (hunk 4 above)
Comment (jadolg) [reply tests/test_settings.py]: Oh, I see now why codecov was complaining. Yes, that's enough.

**Comment by jadolg** [comment]
> sorry I totally forgot about this PR... :/

Not a problem :) Thanks for contributing. I'll merge and release today.

**Review by jadolg — APPROVED** [review]

**Pull request merged**